torch/csrc/jit/frontend/type_refinement.h

```diff
@@ -0,0 +1,177 @@
+namespace torch {
```

Comment:
Also, it is weird that aliases like TypePtr are being used here but there are no includes. I'm guessing that everything works because this file is always included after a header that directly or indirectly includes its definition.